aws/data_source_aws_db_subnet_group_test.go

```diff
@@ -0,0 +1,59 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAWSDbSubnetGroupDataSource_basic(t *testing.T) {
+	rInt := acctest.RandInt()
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSDBSubnetGroupDataSourceConfig(rInt),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "arn"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "description"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "name"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "status"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "subnets.0"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "subnets.1"),
+					resource.TestCheckResourceAttrSet("data.aws_db_subnet_group.bar", "vpc_id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccAWSDBSubnetGroupDataSourceConfig(rInt int) string {
+	return fmt.Sprintf(`
+data "aws_availability_zones" "available" {
```

Comment:
here we should ensure the subnet group's subnets have the same availability_zone group (an error is thrown in the case that they may be different) so an alternative pattern to use instead is
```
data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}
```